zeppelin-server/src/main/java/org/apache/zeppelin/service/ShiroAuthenticationService.java

```diff
@@ -407,8 +407,8 @@ public boolean isAuthenticated() {
     Connection con = null;
     PreparedStatement ps = null;
     ResultSet rs = null;
-    DataSource dataSource = null;
-    String authQuery = "";
+    DataSource dataSource;
+    String authQuery;
```

Comment:
Only change these 2 variables ?

Reply:
In the case of `con`, `ps`, and `rs` variables, they are used in the `finally` block, so if not initialized, a compilation error occurs.